css/gallery.css

```diff
@@ -0,0 +1,111 @@
+/*class for the section of gallery*/
+.container-componet {
+  padding: 60px 0;
+}
+.gallery {
+  width: 635px;
+  height: 88px;
+  font-family: Roboto;
+  font-size: 16px;
+  color: black;
+}
+.gallery__info {
+  top: 0;
+  left: 0;
+  margin-right: 150px;
+  margin-left: 150px;
+  margin-bottom: 17px;
+}
+.text-p {
+  padding-bottom: 17px;
+  margin-block-start: 0px;
+  margin-block-end: 0px;
+}
+.gallery__items {
+  display: flex;
+  flex-flow: row wrap;
+  justify-content: center;
+}
+.gallery__items :nth-child(4),
+.gallery__items :nth-child(8) {
+  margin-right: 0px;
+}
+.gallery-item {
+  background-position: center;
+  background-size: cover;
+  position: relative;
+  width: 227px;
+  height: 227px;
+  margin-right: 17px;
+  margin-bottom: 17px;
+  box-shadow: 0px 0px 2px 0px rgba(0, 0, 0, 0.75);
+  overflow: hidden;
+}
+.gallery-item__images {
+  width: 100%;
+  height: 100%;
+  background-size: cover;
+  transition: 0.1s;
+  -webkit-transition: all 0.1s ease;
+  -moz-transition: all 0.9s ease; /* Firefox */
+  -o-transition: all 0.9s ease; /* IE 9 */
+  -ms-transition: all 0.9s ease; /* Opera */
+}
+.gallery-item:hover .gallery-item__images {
+  -webkit-transform: scale(1.25);
+  -moz-transform: scale(1.25);
+  -ms-transform: scale(1.25);
+  -o-transform: scale(1.25);
+  transform: scale(1.25);
+}
+/* overlay gray for the images */
+.gallery-item__overlay {
+  position: absolute;
+  bottom: 0%;
+  left: 0%;
+  width: 0%;
+  height: 0%;
+  background-color: rgba(0, 0, 0, 0.541);
+  overflow: hidden;
+  vertical-align: middle;
+}
+.overlay__image {
+  display: block;
+  width: 50px;
+  height: 50px;
+  margin: auto;
+  margin-top: 89px;
+}
+.gallery-item:hover .gallery-item__overlay {
+  width: 227px;
+  height: 227px;
+  cursor: pointer;
+}
+/* start media queries */
+@media screen and (max-width: 960px) {
+  .gallery__info {
+    display: flex;
+    flex-flow: row wrap;
+    margin-left: 10px;
+    margin-right: 10px;
+    margin-bottom: 38px;
+  }
+  .gallery__items {
+    max-width: 960px;
+    margin-right: 10px;
+    margin-left: 10px;
+  }
+  .gallery-item {
+    width: 150px;
+    height: 150px;
+    margin-right: 10px;
+    margin-bottom: 10px;
+  }
+  .gallery-item__overlay {
+    display: none;
+  }
+  .gallery__items :nth-child(4),
+  .gallery__items :nth-child(8) {
```

Comment:
what happens if I have 20 images? are you sure that give constants number is the best approach for add styles?

Reply:
Although we expect everyone to do some research and play around with their code this case is very specific and could lend itself to building super complex solutions to support a "what if?" case.
Remember good code is simple code, if you project doesn't need or is not expected to have 20 images, but you know 8 will be there, make those 8 work perfectly and then give a quick analysis to other scenarios, after that only bring in solutions than do not make your code much more complex.

===

Aunque esperamos que todos hagan invesitgaciones y jueguen con su codigo para aprender, este caso especifico se presta mucho para construir soluciones hiper complicadas para soportar un caso de "que tal si?".
Recuerden que un buen codigo es un codigo sencillo, si su proyectono necesita o no espera tener 20 imagenes, pero saben que si o si tendra 8, hagan que esas 8 funcionen bien, luego analisen eso escenarios extra e incorporen soluciones que no agreguen mayor complejidad a su codigo.

siguiendo con eso, aqui esta como aplicar la propiedad cada 4 elementos si importar cuantos haya, es sencillo pero a mi parecer no tan facil de encontrar cuando aun no tienes mucha idea de que buscas.

```
.gallery__items :nth-child(4n) {
// Styling for every third element here.
}
```

Reply:
can you explainme what is that? 
:nth-child(4n)

Reply:
@Sonny97 please view my comment right before yours, the explanation is there.